appcues/src/main/java/com/appcues/debugger/DebuggerViewModel.kt

```diff
@@ -156,11 +154,16 @@ internal class DebuggerViewModel(override val scope: AppcuesScope, debugMode: De
         }
     }
 
-    fun onDismissAnimationCompleted() {
+    fun onDismissAnimationCompleted(debuggerState: MutableDebuggerState) {
         if (_uiState.value is Dismissing) {
             _uiState.value = Dismissed(uiState.value.mode)
             viewModelScope.cancel()
         }
+
+        // fab is completely dismissed, we can take screenshot
+        if (_uiState.value is Expanded && _uiState.value.mode is ScreenCapture) {
+            captureScreen(debuggerState)
+        }
```

Comment:
I'm assuming this all probably works fine, but I'm having a hard time following exactly really - this seems pretty subtle now. Let me see if my understanding is correct

1. when screen capture starts, it hides the FAB, this happens on L179 in DebuggerComposition - however, this no longer directly triggers the VM to sart the screen capture
2. here, on L164, we check for the dismissal complete, and if in a specific state (expanded in ScreenCapture mode) - we trigger the capture, a side-effect of the dismissal at this point
3. additional debuggerState is passed through the dismissal callbacks to make that available here for `captureScreen(state)`
4. we no longer need prepare(view) in ScreenCaptureProcessor, since it is already hidden. restore is removed as well - is this an issue? When does the FAB come back after the screen capture? The previous logic was to return to the camera icon state so you could take more captures.

Reply:
1,2 ,3 - you go it

4. you can still do it, the prepare and restore was used to  basically remove the view for the screen capturing, but what is actually returning it to the FAB (Idle) is what happens after the dialog is dismissed.


https://github.com/appcues/appcues-android-sdk/assets/5244805/61b8721f-2c5e-4cc1-86bf-b8ced36a5e1d

Reply:
ok next question i would have - does the `R.id.appcues_debugger_view` show up in the screen capture layout structure now? previously, that view was hidden so it would get skipped in the layout hierarchy capture. But now it is still there right? just the composition within it is dismissed?